tests/system/Helpers/URLHelper/CurrentUrlTest.php

```diff
@@ -167,12 +172,11 @@ public function testUriStringNoTrailingSlashAbsolute()
         $_SERVER['REQUEST_URI'] = '/assets/image.jpg';
 
         $this->config->baseURL = 'http://example.com';
-        $request               = Services::request($this->config);
-        $request->uri          = new URI('http://example.com/assets/image.jpg');
 
-        Services::injectMock('request', $request);
+        $uri = 'http://example.com/assets/image.jpg';
+        $this->setService($uri);
 
-        $this->assertSame('/assets/image.jpg', uri_string());
+        $this->assertSame('assets/image.jpg', uri_string());
```

Comment:
I'm confused why this changed just due to our fixtures. Do you know the explanation?

Reply:
Probably the test code was wrong.
The process that detects the URI is complex, and the previous test had incomplete setup.

Reply:
The new version looks correct so I'm gonna roll with it. URI is so unnecessarily complex 😕